src/bacnet/basic/service/s_delete_object.c

```diff
@@ -53,12 +53,21 @@ uint8_t Send_Delete_Object_Request(
     BACNET_DELETE_OBJECT_DATA data = { 0 };
     BACNET_NPDU_DATA npdu_data = { 0 };
     uint8_t service = SERVICE_CONFIRMED_DELETE_OBJECT;
+#if BACNET_SEGMENTATION_ENABLED
+    uint8_t segmentation = 0;
+    uint16_t maxsegments = 0;
+#endif
 
     if (!dcc_communication_enabled()) {
         return 0;
     }
     /* is the device bound? */
-    status = address_get_by_device(device_id, &max_apdu, &dest);
+    status = address_get_by_device(
```

Comment:
this change is not needed if `address_segment_get_by_device()` is added to address.c module and address_get_by_device() remains unchanged